### PR TITLE
Handle deleted events in eventsByPersistenceId

### LINKED ISSUE #103: Event deletion
Is already in place for replays, but not for by persistence-id queries. For eventsByTag I don't think we will care about it though.

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #103

### CODE REVIEW

File: core/src/test/scala/akka/persistence/couchbase/EventsByPersistenceIdSpec.scala [hunk 1]
```diff
@@ -109,365 +151,414 @@ class EventsByPersistenceIdSpec
...
-        probe
-          .expectNoMessage(noMsgTimeout)
-          .request(5)
-          .expectNext("e-3")
-          .expectComplete() // e-4 not seen
-      }, readOurOwnWritesTimeout)
+      persistentActor ! s"$pid-4"
+      probe.expectMsg(s"$pid-4-done")
+
+      // FIXME this sometimes sees completion instead of that third element
```
Comment (johanandren) [inline core/src/test/scala/akka/persistence/couchbase/EventsByPersistenceIdSpec.scala]: This is #110

File: core/src/test/scala/akka/persistence/couchbase/EventsByPersistenceIdSpec.scala [hunk 2]
```diff
@@ -109,365 +151,414 @@ class EventsByPersistenceIdSpec
...
           .request(100)
-          .expectNextN((1 to 15).map(i => s"i2-$i"))
+          .expectNextN((1 to 15).map(i => s"$pid-$i"))
           .expectComplete()
       }
     }
 
-    "find all events when PersistAll spans partition boundaries" in {
-      val ref = setup("i3", 10)(identity)
+    "find all events when PersistAll spans partition boundaries" in new Setup {
```
Comment (patriknw) [inline core/src/test/scala/akka/persistence/couchbase/EventsByPersistenceIdSpec.scala]: partition boundaries in Couchbase?

File: core/src/test/scala/akka/persistence/couchbase/EventsByPersistenceIdSpec.scala [hunk 3]
```diff
@@ -109,365 +151,414 @@ class EventsByPersistenceIdSpec
...
+          .expectNext((pid, 1, Offset.sequence(1)), (pid, 2, Offset.sequence(2)), (pid, 3, Offset.sequence(3)))
           .expectComplete()
       }
     }
 
-    "produce correct sequence of events across multiple partitions" in {
-      setup("i", 20) { _ =>
-        val src = queries.currentEventsByPersistenceId("i", 0L, Long.MaxValue)
+    // these talk about partitions, are they straight copy pasta from cassandra? We don't have partitions.
+    "produce correct sequence of events across multiple partitions" in new Setup {
```
Comment (patriknw) [inline core/src/test/scala/akka/persistence/couchbase/EventsByPersistenceIdSpec.scala]: partitions?

Comment (johanandren) [reply core/src/test/scala/akka/persistence/couchbase/EventsByPersistenceIdSpec.scala]: Yeah, you see the comment above, the tests already contained these, I'll have to ask @chbatey

**Pull request merged**